internal/transport/http2_client.go

```diff
@@ -1366,7 +1367,9 @@ func (t *http2Client) operateHeaders(frame *http2.MetaHeadersFrame) {
 	if !isGRPC || httpStatusErr != "" {
 		var code = codes.Internal // when header does not include HTTP status, return INTERNAL
 
-		if httpStatusCode != nil {
+		if rawStatusCode != nil {
+			code = *rawStatusCode
```

Comment:
So now here is the big question: what if `rawStatusCode` is `OK`?

If you read the spec literally, we should ignore the HTTP status in this case.  This appears to be what the other languages do as well, so I suppose we should do that, even though to me it appears to be covering up errors in the system.  (@ejona86 @markdroth FYI)

Reply:
As you mentioned - this results in potentially covering up an error. [Here's an example in tests in this PR](https://github.com/grpc/grpc-go/blob/3e948ab878a5ec3670149edd891f9ab625ea9d72/internal/transport/transport_test.go#L2033-L2035).

Reply:
I don't think this part of the change is right.  A missing `content-type` error should continue to take priority over a `grpc-status` of `OK` (though that's admittedly a gray area and also unlikely to happen).  We should still use that precedence, but only change it so we take `grpc-status` over `http-status` if everything higher priority is okay.

Reply:
I took a stab at this. I'm not 100% sure it's what you meant but I think it is. If we receive a grpc-status but have encountered another error, we'll take the http-status over the grpc-status. Otherwise, we'll default to the http-status.